src/store/wallet.js

```diff
@@ -76,19 +76,16 @@ export default {
     setGeneratedTransaction(state, { transaction }) {
       state.generatedTransaction = transaction
     },
-    setAddresses(state, { addresses, address }) {
-      const addressesMap = new Map([])
-      const addAddress = addr => addressesMap.set(addr.address, addr)
-      state.addresses.map(addAddress)
-
-      if (addresses) {
-        addresses.map(addAddress)
-      }
-
+    setNewAddress(state, { address }) {
```

Comment:
What do you think about rename it as `addAddress`
```suggestion
    addAddress (state, { address }) {
```